### DISCUSSION TITLE
PNG conversion resulting in black background instead of transparent

### DISCUSSION THREAD

**Discussion by brockshankar** [ImageMagick/ImageMagick#3964] [Help]
I am unable to convert the PNG from 32bit to 8 bit. I used photoshop and was able to. I need to use IM to convert it. am i missing any option?

I am using "magick convert test.png  -colors 256 PNG8:convert_IM.png" . I was able to convert other images, i believe the transparency is having an issue.  Can anyone let me know if i am missing any options. Attached original"test.png", photoshop converted and IM converted 8 bit PNG image.
[PNGconvert.zip](https://github.com/ImageMagick/ImageMagick/files/6842838/PNGconvert.zip)

**Comment by snibgo** [comment]
> I am unable to convert the PNG from 32bit to 8 bit.

But what does that mean? Perhaps you mean the input has 4 channels, each with 8 bits. And you want to reduce that to 8 bits total. If so, then your command is correct, and creates an indexed (palette) file.

Or perhaps you want to reduce it to one channel, with 8 bits in that channel.

Or something else.

**Reply by brockshankar** [reply]
I used the same command on multiple files and was able to convert. But, it doesn't work in the file i attached. I believe it is due to it being transparent. I also attached another image which i converted in photoshop.

**Comment by snibgo** [comment]
The Photoshop version is palette-alpha. If that is what you want, then:
```
magick test.png -type PaletteAlpha p8.png
```

**Reply by fmw42** [reply]
This works for me.

`magick test.png +dither -colors 256 -type PaletteAlpha test8.png`

![test8](https://user-images.githubusercontent.com/13458218/126206789-275c12d1-2d8d-4506-8c5c-625a539bd8ff.png)

**Reply by brockshankar** [reply]
@fmw42 @snibgo  That works. Thank you.
